src/Compilers/CSharp/Test/Emit3/FlowAnalysis/StructTests.cs

```diff
@@ -219,6 +219,196 @@ public void GenericStructWithPropertyUsingStruct()
                 Diagnostic(ErrorCode.ERR_StructLayoutCycle, "P").WithArguments("S<T>.P", "S<T[]>?").WithLocation(3, 13));
         }
 
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/66844")]
+        public void InstanceMemberExplosion_01()
```

Comment:
> Consider testing ref structs with ref fields.

Wy do you think this will be interesting in context of this PR?

Reply:
Ref fields cannot cause cycles in the struct layout, but I think an error would be reported today. However, since ref to ref struct is an error anyway, it's probably fine, and unrelated to the PR.

Reply:
In my opinion this scenario is not interesting for the purpose of this PR regardless whether ref fields are considered as a possible source of cycle. Because this PR doesn't change what fields are looked at. It simply ensures we do not infinitely expand types in the process.